crates/ruff_linter/src/rules/ruff/rules/unnecessary_regular_expression.rs

```diff
@@ -248,3 +249,28 @@ impl<'a> ReFunc<'a> {
         })
     }
 }
+
+/// Recursively try to resolve `name` to an [`ExprStringLiteral`] in `semantic`.
+fn resolve_name<'a, 'b>(
+    name: &'a Expr,
+    semantic: &'b SemanticModel,
+) -> Option<&'b ExprStringLiteral>
+where
+    'a: 'b,
+{
+    if name.is_string_literal_expr() {
+        return name.as_string_literal_expr();
+    }
+
+    let mut name = name;
+    while let Some(name_expr) = name.as_name_expr() {
+        let binding = semantic.binding(semantic.only_binding(name_expr)?);
+        let value = find_binding_value(binding, semantic)?;
+        if value.is_string_literal_expr() {
+            return value.as_string_literal_expr();
+        }
+        name = value;
+    }
+
```

Comment:
Does the following of aliases find many new diagnostics? 

I'm otherwise leaning towards not supporting it. It seems rare that you have

```py
pat = "string"
alias = pat

re.sub(alias, ...)
```

Reply:
It looks like there's nothing new from the ecosystem check, which surprised me. I'm happy not supporting this either, in light of that. The literal case definitely seems like the main one.

Reply:
Let me know what you decide or when it's ready to merge.

Reply:
Do you think it's worth following any variables? I can remove the loop to follow only a direct reference, or we can even close the whole PR if you think the literal case is all we need. I'm happy with any of these options.

Reply:
Oh, I like that the rule now follows a direct reference. We should merge the PR. The only part I'm unsure about is whether it's worth to follow a chain of aliases.

Reply:
Okay, I think I'll go with just the direct reference. I was a little worried this could somehow become an infinite loop anyway.